data/service/api/v1/summary.go

```diff
@@ -0,0 +1,113 @@
+package v1
+
+import (
+	"net/http"
+    "strconv"
+
+	//"github.com/tidepool-org/platform/data"
+	dataService "github.com/tidepool-org/platform/data/service"
+	//"github.com/tidepool-org/platform/page"
+	//"github.com/tidepool-org/platform/permission"
+	"github.com/tidepool-org/platform/request"
+)
+
+func SummaryRoutes() []dataService.Route {
+	return []dataService.Route{
+		dataService.MakeRoute("GET", "/v1/summary/:userId", Authenticate(GetSummary)),
+		dataService.MakeRoute("POST", "/v1/summary/:userId", Authenticate(UpdateSummary)),
+		dataService.MakeRoute("GET", "/v1/agedsummaries/:minutes", Authenticate(GetAgedSummaries)),
+	}
+}
+
+func GetSummary(dataServiceContext dataService.Context) {
+	res := dataServiceContext.Response()
+	req := dataServiceContext.Request()
+	dataClient := dataServiceContext.DataClient()
+
+	//details := request.DetailsFromContext(req.Context())
+	/*if details == nil {
+		request.MustNewResponder(res, req).Error(http.StatusUnauthorized, request.ErrorUnauthenticated())
+		return
+	}*/
+
+	responder := request.MustNewResponder(res, req)
+
+	id := req.PathParam("userId")
+	if id == "" {
+		responder.Error(http.StatusBadRequest, request.ErrorParameterMissing("id"))
+		return
+	}
+
+	summary, err := dataClient.GetSummary(req.Context(), id)
```

Comment:
I think this will return a mostly empty `Summary` object (the NoDoc branch in `dataClient.GetSummary`) **if** there's never been a calculation done on that user's account. Granted, it's a transient corner case for new accounts and accounts that we haven't "refreshed" yet.

Should the API instead return 204, 404 or something equally explicit to say "don't have it yet"?

Reply:
Would it be okay practice to return 201 or 202 here?
The intent was to create the object for a future pass, to start calculating summaries on first query, to prevent any unnecessary calculation. (clinic-less users)
Clients can detect a 0 datestamped summary as "no data", but if we rather the clients check for return codes, we could return 201 or 202.

Reply:
Made it return 204 if summary returned has 0000-00-00 datestamp, this should also cover for the case of repeated calls which do return a value from the db, but are not ready yet.